go/vt/vtgate/planbuilder/postprocess.go

```diff
@@ -73,8 +73,15 @@ func pushOrderBy(orderBy sqlparser.OrderBy, bldr builder) error {
 	s := bldr.Symtab().SetState(symtabOrderBy)
 	defer bldr.Symtab().SetState(s)
 
-	if orderBy == nil {
+	switch len(orderBy) {
+	case 0:
```

Comment:
This is actually what you want to do in go, otherwise you have multiple ifs:
if len(orderby) == 0 {
} else if len(orderby) == 1 {
}
Basically 'else if' is not encouraged at all.